File: exercism/ocaml/rna-transcription/dna.ml
open Core.Std

type dna = [ `A | `C | `G | `T ]
type rna = [ `A | `C | `G | `U ]

let to_rna : dna list -> rna list =
  let f = function
    | `A -> `U
    | `T -> `A
    | `C -> `G
    | `G -> `C
  in
  List.map ~f

File: exercism/haskell/rna-transcription/src/DNA.hs
module DNA (toRNA) where

toRNA :: String -> Maybe String
toRNA = sequence . map convert
  where
    convert 'C' = Just 'G'
    convert 'G' = Just 'C'
    convert 'A' = Just 'U'
    convert 'T' = Just 'A'
    convert  _  = Nothing
